qemu/tests/luks_image_over_qsd.py

```diff
@@ -0,0 +1,92 @@
+        6.2 luks key management.
+        :param test: VT test object.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+    qsd = QsdDaemonDev("qsd1", params)
+    qsd.start_daemon()
+
+    luks_image_size = params['image_size_stg1']
+    luks_img_param = params.object_params('stg1')
+    luks_img = qemu_storage.QemuImg(luks_img_param, data_dir.get_data_dir(),
+                                    'stg1')
+    nbd_image_tag = params['nbd_image_tag']
+    nbd_image_params = params.object_params(nbd_image_tag)
+    localhost = socket.gethostname()
+    nbd_image_params['nbd_server'] = localhost if localhost else 'localhost'
+    qemu_img = qemu_storage.QemuImg(nbd_image_params, None, nbd_image_tag)
+    qemu_img.info()
+
+    params["start_vm"] = 'yes'
+
+    login_timeout = params.get_numeric("login_timeout", 360)
+    env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+    output = session.cmd_output("lsblk -nd | grep '[s|v]db'")
+    if not output:
+        test.fail("There is no the luks image in guest, check please")
+    elif luks_image_size not in output:
+        test.fail(
+            "The image size is not matched to the data disk in guest, "
+            "check please.")
+    session.close()
+    vm.destroy()
+
+    # query-block-exports
+    out = qsd.monitor.cmd("query-block-exports")[0]
+    if out.get("type") != "nbd":
+        test.fail("The exported type is not matched to 'nbd'.")
+    out = qsd.monitor.cmd("query-named-block-nodes")[0]
+    image_info = out.get("image")
+    image_size = utils_numeric.normalize_data_size(
+        str(image_info.get("virtual-size")), "G").split('.')[0] + 'G'
+    image_name = image_info.get("filename")
+    if image_size != luks_image_size or image_name != luks_img.image_filename:
+        test.fail(
+            "The image size(%s) or image name(%s) is not matched to the "
+            "original image." % (image_size, image_name))
+
+    # luks key management
+    args1 = {"qom-type": "secret", "id": "sec1", "data": "redhat1"}
+    args2 = {"qom-type": "secret", "id": "sec2", "data": "redhat2"}
+    for args in [args1, args2]:
+        out = qsd.monitor.cmd("object-add", args)
+        if "error" in out:
+            test.fail("Add secret object failed, check please")
+    # add a new password to a free keyslot
+    args = {"node-name": "fmt_stg1", "job-id": "job_add_key1",
+            "options": {"driver": "luks", "state": "active",
+                        "new-secret": "sec1", "keyslot": 1, "iter-time": 10}}
+    try:
+        qsd.monitor.cmd("x-blockdev-amend", args)
+    except QMPCmdError as e:
+        qmp_error_msg = params.get("qmp_error_msg")
+        if qmp_error_msg not in str(e.data):
+            test.fail("The error msg(%s) is not correct." % str(e))
+
```

Comment:
add else clause here
                else:
                         test.fail("Unexpected success when running x-blockdev-amend")
because we expect an error here, right?